src/LinuxTracing/ThreadDepedencyTestApp.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <absl/synchronization/mutex.h>
+
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <iostream>
+#include <iterator>
+#include <mutex>
+#include <thread>
+
+#include "OrbitBase/ThreadUtils.h"
+
+static std::atomic<bool> exit_requested;
+static std::atomic<int> num;
+
+absl::Mutex test_mutex;
+
+constexpr int kNumOfThreads = 4;
+constexpr int kSigint = 2;
+
+std::thread my_threads[kNumOfThreads];
+
+void SignalHandler(int signum) {
+  std::cout << "Cleaning up!" << std::endl;
+  exit_requested.store(false);
```

Comment:
I think you don't need to use `load` and `store`. You can use the variable directly and it will be cast to its content.